beacon/blockchain/execution_engine.go

```diff
@@ -46,69 +47,9 @@ func (s *Service[
 		return
 	}
 
-	if !s.shouldBuildOptimisticPayloads() && s.localBuilder.Enabled() {
-		s.sendNextFCUWithAttributes(ctx, st, blk, lph)
-	} else {
-		s.sendNextFCUWithoutAttributes(ctx, blk, lph)
-	}
-}
-
-// sendNextFCUWithAttributes sends a forkchoice update to the execution
-// client with attributes.
-func (s *Service[
-	_, _, ConsensusBlockT, _, _, BeaconStateT, _,
-	_, _, ExecutionPayloadHeaderT, _, _, _, _,
-]) sendNextFCUWithAttributes(
-	ctx context.Context,
-	st BeaconStateT,
-	blk ConsensusBlockT,
-	lph ExecutionPayloadHeaderT,
-) {
-	beaconBlk := blk.GetBeaconBlock()
-
-	stCopy := st.Copy()
-	if _, err := s.stateProcessor.ProcessSlots(
-		stCopy, beaconBlk.GetSlot()+1,
-	); err != nil {
-		s.logger.Error(
-			"failed to process slots in non-optimistic payload",
-			"error", err,
-		)
-		return
-	}
-
-	nextPayloadTime := payloadtime.Next(
-		blk.GetConsensusTime(),
-		lph.GetTimestamp(),
-		true, // buildOptimistically
-	).Unwrap()
-
-	// We set timestamp check on Bartio for backward compatibility reasons
-	// TODO: drop this we drop other Bartio special cases.
-	if s.chainSpec.DepositEth1ChainID() == spec.BartioChainID {
-		nextPayloadTime = max(
-			//#nosec:G701
-			uint64(time.Now().Unix()+1),
-			uint64((lph.GetTimestamp() + 1)),
-		)
-	}
-
-	prevBlockRoot := beaconBlk.HashTreeRoot()
-	if _, err := s.localBuilder.RequestPayloadAsync(
-		ctx,
-		stCopy,
-		beaconBlk.GetSlot()+1,
-		nextPayloadTime,
-		prevBlockRoot,
-		lph.GetBlockHash(),
-		lph.GetParentHash(),
-	); err != nil {
-		s.logger.Error(
-			"failed to send forkchoice update with attributes in non-optimistic payload",
-			"error",
-			err,
-		)
-	}
+	// Send a forkchoice update without payload attributes to notify
+	// EL of the new head.
+	s.sendNextFCUWithoutAttributes(ctx, blk, lph)
```

Comment:
another q: `sendNextFCUWithoutAttributes` can err, in which case we only log the error.
Should we return error instead and, pending the goroutine remova, err on block finalization if we fail to ping the EVM? @nidhi-singh02  @shotes

Reply:
My understanding is that we should **not** err over non-deterministic errors like this in the finalize block context, just double-checking here

Reply:
In the short term, I think this PR lgtm when it comes to fixing the initial issue.

But @abi87 brings up good questions that I think deserve a lot of attention and potentially some reworking - particularly in how beacon-kit handles staying in sync with its execution client. Here are some questions that I think will drive this discussion:
- What happens when `sendNextFCUWithoutAttributes()` fails because of some ephemeral issue in trying to send the forkchoiceUpdated RPC call? like a onetime networking error
- What happens when `sendNextFCUWithoutAttributes()` continues to fail over and over from network issues? or the EL client is syncing?
- What happens when `sendNextFCUWithoutAttributes()` fails because of an invalid parent block hash? Or an invalid forkchoice state? (I don't think this should *ever* happen in `FinalizeBlock()`, and if it does, there is clearly some chain-breaking bug somewhere.)

In my opinion, beacon-kit cannot simply move on with processing blocks if the forkchoiceUpdate is failing for any reason. At the same time, it is difficult to suggest returning error in `FinalizeBlock()`. A potential path forward would be to start the FCU in a goroutine that:
- retries on ephemeral error, with an exponential backoff that will timeout eventually
- errors when there is a true failure
- prevents the beacon-kit node from participating (not proposing, verifying proposals, or processing new blocks) until the forkchoice is updated on the EL.

Reply:
@shotes fine with merging this PR and tackle the retry issues separately, but please make an issue so that we can better track this.
You extended comment may be the issue content.
Once that is done, I am down to merge this

Reply:
New issue created in https://github.com/berachain/beacon-kit/issues/2267